src/pages/permissions/PermissionIdentities.tsx

```diff
@@ -0,0 +1,251 @@
+import { Link, useSearchParams } from "react-router-dom";
+import { queryKeys } from "util/queryKeys";
+import useSortTableData from "util/useSortTableData";
+import PermissionIdentitiesFilter, {
+  AUTH_METHOD,
+  PermissionIdentitiesFilterType,
+  QUERY,
+} from "./PermissionIdentitiesFilter";
+import { useSettings } from "context/useSettings";
+import PermissionIdentityEditGroupsBtn from "./PermissionIdentityEditGroupBtn";
+import { encodeIdentityNameForUrl } from "util/permissionIdentities";
+import usePanelParams, { panels } from "util/usePanelParams";
+import CustomLayout from "components/CustomLayout";
+import PageHeader from "components/PageHeader";
+import NotificationRow from "components/NotificationRow";
+import HelpLink from "components/HelpLink";
+import { useDocs } from "context/useDocs";
+import EditIdentityGroupsPanel from "./panels/EditIdentityGroupsPanel";
+
+const PermissionIdentities: FC = () => {
+  const notify = useNotify();
+  const {
+    data: identities = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.identities],
+    queryFn: () => fetchIdentities(),
+  });
+  const { data: settings } = useSettings();
+  const docBaseLink = useDocs();
+  const panelParams = usePanelParams();
+  const [searchParams] = useSearchParams();
+  const [selectedIdentityNames, setSelectedIdentityNames] = useState<string[]>(
+    [],
+  );
+
+  if (error) {
+    notify.failure("Loading identities failed", error);
+  }
+
+  const headers = [
+    { content: "Name", className: "name", sortKey: "name" },
+    { content: "ID", sortKey: "id" },
+    { content: "Auth method", sortKey: "authmethod" },
+    { content: "Type", sortKey: "type" },
+    {
+      content: "Groups",
+      sortKey: "groups",
+      className: "u-align--right",
+    },
+  ];
+
+  const filters: PermissionIdentitiesFilterType = {
+    queries: searchParams.getAll(QUERY),
+    authMethod: searchParams.getAll(AUTH_METHOD),
+  };
+
+  const filteredIdentities = identities.filter((identity) => {
+    if (
+      !filters.queries.every(
+        (q) =>
+          identity.name.toLowerCase().includes(q) ||
+          identity.id.toLowerCase().includes(q),
+      )
+    ) {
+      return false;
+    }
+
+    if (
+      filters.authMethod.length > 0 &&
+      !filters.authMethod.includes(identity.authentication_method)
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+
+  const rows = filteredIdentities.map((identity) => {
+    const isLoggedInIdentity = settings?.auth_user_name === identity.id;
+    const isTlsIdentity = identity.authentication_method === "tls";
+    return {
+      name: isTlsIdentity ? "" : identity.name,
+      className: "u-row",
+      columns: [
+        {
+          content: (
+            <Link
+              className="u-truncate name-link"
+              title={`name: ${identity.name}`}
+              to={`/ui/permissions/identity/${identity.authentication_method}/${encodeIdentityNameForUrl(identity.name)}`}
```

Comment:
```suggestion
              to={`/ui/permissions/identity/${identity.authentication_method}/${identity.id}`}
```
We can't use the name, because it is not guaranteed to be unique. We can have two TLS certs with the same names.

Reply:
I would still keep the `encodeIdentityNameForUrl` function because identity id can be an email.